tests/test_database.py

```diff
@@ -238,3 +238,16 @@ def test_insert_values_to_table():
     db.insert_values_to_table(table, values)
     assert "INSERT INTO a_table" in db.captured_queries[0]
     assert values == db.captured_multiparams[0][0]
+
+
+def test_list_data_models():
```

Comment:
The most complex logic in this PR is implemented in the `list_data_models` and `list_datasets` methods in database.py. Yet there are no integration tests for these methods. 

In my opinion this complex logic must move to the corresponding use cases and integration tests should be added there. The database methods `get_datasets` and `get_data_models` should also be tested with integration tests and possible with unit tests.